### DISCUSSION TITLE
Future date in LICENSE.md

### DISCUSSION THREAD

**Discussion by scaleoutsean** [dragonflydb/dragonfly#301] [Q&A] [answered]
`<u>Change Date</u>: June 1, 2027`
https://github.com/dragonflydb/dragonfly/blame/main/LICENSE.md#L9

I haven't seen a future-date license. What does that mean? Will you change the license in 2027?

**Comment by romange** [comment, marked as answer]
Actually, there are quite a few projects that use BSL. 
Check out, for example, https://www.cockroachlabs.com/docs/stable/licensing-faqs.html#license-conversion-timeline from CockroachDB.

it means that the license guarantees that the current code in the repo becomes apache licensed in 2027.
We, of course, reserve the right to update conversion dates for our future versions.

Having said that, as long as you do not plan to provide managed services with Dragonfly or any other business activity by selling Dragonfly as a service - you can use Dragonfly for your internal needs for free. This license is considered to be better for commercial companies than, say, open-source licenses like GPL or AGPL.

**Reply by scaleoutsean** [reply]
Ok, "change date" in hindsight makes sense. "Conversion date" would be better and probably given me a hint to search for similar clauses in other licenses.
